src/solver/ts-generator/thermal.cpp

```diff
@@ -596,7 +597,81 @@ void GeneratorTempData::operator()(Data::Area& area, Data::ThermalCluster& clust
 }
 } // namespace
 
+
+std::vector<std::pair<std::string, std::string>> splitStringIntoPairs (const std::string& s,
+                                                                       char delimiter1,
+                                                                       char delimiter2)
+{
+    std::vector<std::pair<std::string, std::string>> pairs;
+    std::stringstream ss(s);
+    std::string token;
+
+    while (std::getline(ss, token, delimiter1))
+    {
+        size_t pos = token.find(delimiter2);
+        if (pos != std::string::npos)
+        {
+            std::string first = token.substr(0, pos);
+            std::string second = token.substr(pos + 1);
+            pairs.push_back(std::make_pair(first, second));
+        }
+        else
+        {
+            logs.warning() << "Error with area and cluster: " << token;
+            logs.warning() << "Correct format: \"area1.cluster1;area2.cluster2\"";
+        }
+    }
+
+    return pairs;
+}
+
+static std::vector<Data::ThermalCluster*> getClustersToGen(Data::AreaList& areas,
+                                                           const std::string& clustersToGen,
+                                                           bool globalThermalTSgeneration,
+                                                           bool refreshTSonCurrentYear)
+{
+
+    std::vector<Data::ThermalCluster*> clusters;
+
+    if (clustersToGen.empty()) // generate TS for all clusters
+    {
+        areas.each([&clusters, &globalThermalTSgeneration, &refreshTSonCurrentYear](Data::Area& area) {
+            for (auto cluster : area.thermal.list.all())
+                if (cluster->doWeGenerateTS(globalThermalTSgeneration) && refreshTSonCurrentYear)
+                    clusters.push_back(cluster.get());
+        });
+    }
+    else
+    {
+        const auto names = splitStringIntoPairs(clustersToGen, ';', '.');
+
+        for (const auto& [areaName, clusterName] : names)
+        {
+            logs.info() << "Generating ts for area: " << areaName << " and cluster: " << clusterName;
+
+            auto* area = areas.find(areaName);
+            if (!area)
+            {
+                logs.warning() << "Area not found: " << areaName;
+                continue;
+            }
+
+            auto* cluster = area->thermal.list.findInAll(clusterName);
+            if (!cluster)
+            {
+                logs.warning() << "Cluster not found: " << clusterName;
+                continue;
+            }
+
+            clusters.push_back(cluster);
+        }
+    }
+
+    return clusters;
+}
+
 bool GenerateThermalTimeSeries(Data::Study& study,
+                               const std::string& clustersToGen,
                                uint year,
                                bool globalThermalTSgeneration,
                                bool refreshTSonCurrentYear,
```

Comment:
Move `getClustersToGen` up one level

```suggestion
bool GenerateThermalTimeSeries(const std::vector<ThermalCluster*> clusters, uint year);
// Call (solver.hxx)
auto clusters = getClustersToGen(/* ... */);
GenerateThermalTimeSeries(clusters, year);
```